model/template.go

```diff
@@ -77,11 +77,150 @@ func fillInTemplate(db *gorm.DB, templateModel *TemplateModel) {
 	if catalog != nil {
 		templateModel.Catalog = catalog.Name
 	}
+
 	templateModel.Categories = lookupTemplateCategories(db, templateModel.ID)
 	templateModel.Labels = lookupTemplateLabels(db, templateModel.ID)
 	templateModel.Versions = lookupVersions(db, templateModel.ID)
 }
 
+func templateCategoryMap(db *gorm.DB, templateIDList []int) map[int][]string {
+	categoriesQuery := `
+	SELECT template_id, category_id, name
```

Comment:
Would be nice to use the table aliases here so you know which table each field comes from.  (`SELECT tc.template_id, tc.category_id, c.name`)